### ISSUE TITLE
Ready Event Emitted on Error?

### ISSUE THREAD

**Issue by OmriAharon** [launchdarkly/js-client#113]
Didn't find an error handling section in the [documentation](https://docs.launchdarkly.com/docs/js-sdk-reference), please let me know if I've missed it.

I'm mocking the call to initialise the client with `500`, but it still seems to emit the `ready` event. Is this by design? How can I differentiate an error and success?

**Comment by eli-darkly** [comment]
As you saw, the `ready` event gets emitted as soon as the initialization logic has finished, regardless of whether it succeeded. Currently the only way to detect failed initialization would be to listen for an `error` event on the client. This will be emitted _before_ the `ready` event, and the type of the error object will be `LaunchDarklyFlagFetchError`. If you do not receive an `error before you get a `ready`, then initialization succeeded.

**Comment by eli-darkly** [comment]
But, this reminded us that we had intended to add a better way of handling this: a method returning a promise that would be resolved on successful initialization, or rejected if initialization failed. We already did that in the Node SDK, but it didn't make it into the client-side JS SDK. So we will be adding that soon.

**Comment by OmriAharon** [comment]
Thanks! @eli-darkly

**Comment by eli-darkly** [comment]
Hi - sorry this took a while, but we have released version 2.7.0 which addresses this issue with a new method, `waitForInitialization`. This method returns a Promise that will be resolved on successful initialization, or rejected on failure (unlike `waitUntilReady()` and the `ready` event, which do not distinguish between success and failure). You would use it like this:

```
    myClient.waitForInitialization().
        then(function() {
            // client initialized successfully
        }).
        catch(function(err) {
            // initialization failed
        });
```

Hope this helps.

**Comment by OmriAharon** [comment]
I wish all other project maintainers regarded 10 days as "too long", would have made my life much easier 😄 

Thanks @eli-darkly!
